controller/volume_controller.go

```diff
@@ -592,8 +597,18 @@ func (vc *VolumeController) cleanupCorruptedOrStaleReplicas(v *longhorn.Volume,
 		}
 
 		if r.Spec.FailedAt == "" {
+			// Cleanup extra failed to scheduled replica
+			// when no eviction requested
+			if healthyCount >= v.Spec.NumberOfReplicas &&
```

Comment:
How about using the following logic:
```
// skip running replicas
if r.Spec.HealthyAt != "" {
    continue
}
// when the volume is not Healthy, a new replica will be still recreated if there is a cleanup
if healthyCount < v.Spec.NumberOfReplicas {
    continue
}
// avoid the clean up during the eviction
if hasEvictionRequestedReplicas {
    continue
}
```

Then you don't need to delete the replica here.

Reply:
Following your logic, I prefer to put them together like this:
`
//comments here
if r.Spec.HealthyAt != "" || healthyCount < v.Spec.NumberOfReplicas || hasEvictionRequestedReplicas {
        continue
}
`
what do you think?

Reply:
Updated this part, please take a look. Thanks.